src/server/repositories/UrlRepository.ts

```diff
@@ -228,14 +229,16 @@ export class UrlRepository implements UrlRepositoryInterface {
     state: string | undefined,
     isFile: boolean | undefined,
   ): Promise<UrlDirectoryPaginated> {
+    // Extract shortUrls with regex
+    const newQuery = this.queryFilter(query)
```

Comment:
you could simply do...
```suggestion
    const newQuery =extractShortUrl(query) || query
```